src/components/Product-Cards/Product-Cards.js

```diff
@@ -0,0 +1,45 @@
+import { useEffect , useState } from 'react';
+import axios from "axios";
+
+
+import {useProduct} from "../../context/Productcontext.js";
+import "./Product-Cards.css";
+export  const ProductCards = () => {
+    const {productState,getData,sortData,filterData} =useProduct()
+    useEffect(()=>{
+       productState.products.length===0 && 
+       getData();
+       console.log(productState);
+    },[])
+    const sortedData=sortData(productState.products,productState.filters.priceSort);
+    const filteredAndSortedData=filterData(sortedData,productState.filters.filterByCategory,productState.filters.filterByRating,productState.filters.filterByRange);
+    console.table(filteredAndSortedData)
+    return (
+      <>
+        <div className="product-listing-container">
+          <div className="product-container">
+            {filteredAndSortedData?.map((product) => (
```

Comment:
You could destructure the names from the object in the array and use them instead of doing  `product.` everytime